burr/core/persistence.py

```diff
@@ -192,6 +195,17 @@ def initialize(self):
         """Creates the table if it doesn't exist"""
         # Usage
         self.create_table_if_not_exists(self.table_name)
+        self._initialized = True
+
+    def is_initialized(self) -> bool:
```

Comment:
MIght be nice to have a comment about why this is backwards compatible/how it works (E.G. why you don't need to call initialize).

Reply:
no you do need to call initialize() at some point, but not in the lifecycle of the object, since it's a DB with state we can check whether things are correct or not. But sure.